src/Util/ClassSourceManipulator.php

```diff
@@ -178,6 +180,40 @@ public function addInterface(string $interfaceName)
         $this->updateSourceCodeFromNewStmts();
     }
 
+    public function addTrait($classTraitName)
+    {
+        $this->addUseStatementIfNecessary($classTraitName);
+
+        $classNode = $this->getClassNode();
+        $name = Str::getShortClassName($classTraitName);
+
+        if (empty($classNode->stmts)) {
+            $classNode->stmts[0] = new Node\Name('use '.$name.';');
+        } else {
+            $traitNode = $this->findFirstNode(function ($node) {
+                return $node instanceof Node\Stmt\Class_
+                    && !empty($node->stmts[0])
+                    && $node->stmts[0] instanceof Node\Stmt\TraitUse;
+            });
```

Comment:
This isn't actually finding a "trait node". This is kind of just checking that the `Class_` node has a `Node\Stmt\TraitUse` as its first statement. This callback will be called for *every* node in the class. And so, I think you can simply do:

```php
return $node instanceof Node\Stmt\TraitUse;
```

Then `$traitNode` will actually be an instance of `TraitUse`.

Reply:
Hmm, but overall, I think we can simplify this and the code below by adding a new `findAllNodes(callable $filterCallback)` method that would return `Node[]`. Then we could find *all* of the `TraitUse` nodes with it, loop over *all* of them, and check if any of them match the one that we're trying to add.

Reply:
Done, and i also add a test : testAddTraitAlReadyExists()